### PR TITLE
Add Http2Client and the session manager and selectors necessary

### PR DESCRIPTION
**Pull request by bryce-anderson**
Finally, we have a HTTP/2 client. It still needs a battery of
end-to-end tests which are run against a local server.

### CODE REVIEW

**Comment by bryce-anderson** [comment]
I think the session managers as a whole need a large amount of work. Specifically, giving some thought to timeouts and failures would be nice.

File: http/src/main/scala/org/http4s/blaze/http/ClientSessionManagerImpl.scala [hunk 1]
```diff
@@ -70,7 +70,7 @@ private final class ClientSessionManagerImpl(sessionCache: java.util.Map[Connect
             h2.closeNow() // make sure its closed
             it.remove()
 
-          case h2: Http2ClientSession if h2.quality > 0.1 =>
+          case h2: Http2ClientSession if 0.1 < h2.quality =>
```
Comment (rossabaker) [inline http/src/main/scala/org/http4s/blaze/http/ClientSessionManagerImpl.scala]: What prompted this flip?

Comment (bryce-anderson) [reply http/src/main/scala/org/http4s/blaze/http/ClientSessionManagerImpl.scala]: Just that I think it reads easier.

Comment (jcranky) [reply http/src/main/scala/org/http4s/blaze/http/ClientSessionManagerImpl.scala]: Maybe while at it, `0.1` could also get a name and be even easier to read?

Comment (bryce-anderson) [reply http/src/main/scala/org/http4s/blaze/http/ClientSessionManagerImpl.scala]: Done.

**Comment by bryce-anderson** [comment]
I think this is about ready, more feedback appreciated. That said, no rush: the server side is what is left and is kind of orthogonal to this.

**Pull request merged**